projects/js-packages/publicize-components/src/components/media-section/index.js

```
@@ -25,6 +25,24 @@ import useSocialMediaConnections from '../../hooks/use-social-media-connections'
 import VideoPreview from '../video-preview';
 import styles from './styles.module.scss';
 
+/**
+ * Get meta data from a VideoPress video.
+ *
+ * @param {object} video - VideoPress media object.
+ * @returns {{mime: string, fileSize: number, length: number}} - Metadata
+ */
+const getVideoPressMetadata = video => {
+	if ( ! video.source_url || ! video?.media_details?.videopress?.duration ) {
+		return {};
+	}
+
+	return {
+		mime: `video/${ video.source_url.split( '.' ).pop() }`,
+		fileSize: 1000000, // 1 MB - TODO: calculate actual file size somehow
```

VideoPress doesn't expose the file size of the original video, so I've just hardcoded a valid value here for now. It will fail silently if the video is too large, which isn't ideal but I guess better than nothing.

I tried fetching the source URL to get the `Content-Length` header but that might cause other issues that we can't fix before the code freeze. There's definitely room for improvement here, especially considering VideoPress converts to multiple formats and sizes already.

---

Just to confirm the tradeoff here is video size validations may not work with Videopress uploads, correct?

---

Yes exactly, they'll always be treated as a valid file size. It should only cause issues for really big videos.